### PR TITLE
Add new member

### PR DESCRIPTION
**Pull request by Chestnut-lol**
Added the following functionalities:
1. Can create new Member by inputting name and email.
2. Enable editting member info.

### CODE REVIEW

File: Bookish/Controllers/MemberController.cs
Comment (emilypatsko) [inline Bookish/Controllers/MemberController.cs:164]: You need a line break between lines 53 and 54

File: Bookish/Controllers/MemberController.cs
Comment (emilypatsko) [inline Bookish/Controllers/MemberController.cs:74]: For clarity, I would put annotations on each method to indicate if they're GET or POST requests (or something else). You can do this by putting e.g. `[HttpGet]` above the name of the method

File: Bookish/Controllers/MemberController.cs
Comment (emilypatsko) [inline Bookish/Controllers/MemberController.cs:166]: Delete commented out code

File: Bookish/Models/Member.cs
Comment (emilypatsko) [inline Bookish/Models/Member.cs:6]: Why is this virtual?

File: Bookish/Controllers/MemberController.cs
Comment (emilypatsko) [inline Bookish/Controllers/MemberController.cs]:
```suggestion
            .Include(m => m.Books)
            .ThenInclude(b => b.Book)
```

File: Bookish/Controllers/MemberController.cs
Comment (emilypatsko) [inline Bookish/Controllers/MemberController.cs]: Line break between lines 66 and 67

Comment (emilypatsko) [inline Bookish/Controllers/MemberController.cs]: I don't see this method being used anywhere?

File: Bookish/Controllers/MemberController.cs
Comment (emilypatsko) [inline Bookish/Controllers/MemberController.cs]: If a member is successfully added, why are we returning an error message? I also think this method needs renaming to be clearer, e.g. `AddNewMember`

File: Bookish/Controllers/MemberController.cs
Comment (emilypatsko) [inline Bookish/Controllers/MemberController.cs]: I think this might break the first time you ever add a member, as there won't be a 'last' member to find the add of and add 1 to.

For future reference, when you build your database code-first, if you give a model an `Id` property, entity framework picks up that it's the primary key and will automatically generate it for you. Read more here: https://docs.microsoft.com/en-us/ef/core/modeling/keys?tabs=data-annotations#configuring-a-primary-key

File: Bookish/Views/Member/AddMember.cshtml
Comment (emilypatsko) [inline Bookish/Views/Member/AddMember.cshtml]:
```suggestion
@Html.TextBoxFor(m => m.Name, new { @class="text", type="name", required="required" })
```

need some more spaces

File: Bookish/Views/Member/EditMember.cshtml
Comment (emilypatsko) [inline Bookish/Views/Member/EditMember.cshtml]: Try not to use inline styles (eg `style="display:inline"`) and instead make use of the classes that Bootstrap has to offer. The documentation is [here](https://getbootstrap.com/docs/5.1/getting-started/introduction/)

File: Bookish/Views/Member/MemberQuery.cshtml
Comment (emilypatsko) [inline Bookish/Views/Member/MemberQuery.cshtml]: Why are there empty paragraphs?

File: Bookish/Views/Member/MemberQuery.cshtml
Comment (emilypatsko) [inline Bookish/Views/Member/MemberQuery.cshtml]: You don't really need an `if` statement with the `foreach` loop - it will loop over all of the items in `Model.Books`, so if there aren't any, the things in the loop just won't execute.

**Review by emilypatsko — CHANGES_REQUESTED** [review]

File: Bookish/Models/Member.cs
Comment (Chestnut-lol) [reply Bookish/Models/Member.cs:6]: This was suggested by Ben since otherwise the databases don't seem to link to each other

File: Bookish/Controllers/MemberController.cs
Comment (Chestnut-lol) [reply Bookish/Controllers/MemberController.cs]: No... but we might use it later...?

File: Bookish/Controllers/MemberController.cs
Comment (Chestnut-lol) [reply Bookish/Controllers/MemberController.cs]: Sorry the naming was really bad - the error message class should have been named a "message" class. It is just a page of plain text with a provided message.

File: Bookish/Models/Member.cs
Comment (BenRamchandani) [reply Bookish/Models/Member.cs:6]: I may well be wrong about this to be honest - it's required for lazy loading, but it looks like that's disabled by default in EF Core. Worth trying to remove it and seeing what happens.

File: Bookish/Controllers/MemberController.cs
Comment (Chestnut-lol) [reply Bookish/Controllers/MemberController.cs]: Yes I have changed the way member Id is generated - it is now the time stamp for the time of creation.

File: Bookish/Models/Member.cs
Comment (Chestnut-lol) [reply Bookish/Models/Member.cs:6]: Ok, we will try it out after we merge together in case of nasty conflicts?

File: Bookish/Bookish.csproj
Comment (emilypatsko) [inline Bookish/Bookish.csproj]: I think these lines got added by accident... you should be able to delete them. Check the project still runs properly when you do!

File: Bookish/Controllers/MemberController.cs
Comment (emilypatsko) [reply Bookish/Controllers/MemberController.cs:74]: @Chestnut-lol Not sure if you didn't see this comment - could you add these annotations to each controller method? Let me know if you'd like to talk more about what they mean :)

File: Bookish/Controllers/MemberController.cs
Comment (emilypatsko) [inline Bookish/Controllers/MemberController.cs:74]: I would rename this method to be more descriptive - what does the method do?

File: Bookish/Controllers/MemberController.cs
Comment (emilypatsko) [inline Bookish/Controllers/MemberController.cs]: I think these should really be `return RedirectToActionResult`s

File: Bookish/Controllers/MemberController.cs
Comment (emilypatsko) [inline Bookish/Controllers/MemberController.cs]: Nice clear method name 😃 

This method is structured well! You should adopt this structure in the rest of your methods. 

- Sanitise inputs
- Get relevant models from inputs
- Check any conditions are met - if not, early return with an error view, or redirect
- Otherwise (all conditions met), do whatever logic you want
- Return the view

File: Bookish/Controllers/MemberController.cs
Comment (emilypatsko) [inline Bookish/Controllers/MemberController.cs]: I think I already commented about this on Blanca's PR:

```suggestion
var resultMember = _dbContext.Members
    .Where(x => x.MemberId == memberId)
    .Include(m => m.Books)
    .ThenInclude(b => b.Book)
    .SingleOrDefault()
```

File: Bookish/Controllers/MemberController.cs
Comment (emilypatsko) [reply Bookish/Controllers/MemberController.cs:166]: Is there a reason you're keeping this commented code in?

File: Bookish/Controllers/MemberController.cs
Comment (emilypatsko) [inline Bookish/Controllers/MemberController.cs:171]: Similar comment as earlier - you shouldn't need to index the results like this. If you're only expecting one model to be returned,  you can use `SingleOrDefault()`

File: Bookish/Controllers/MemberController.cs
Comment (emilypatsko) [reply Bookish/Controllers/MemberController.cs]: Not a reason to add it now - there's a concept in writing good code known as YAGNI (you ain't gonna need it). Take a look [here](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it). Basically, don't add functionality until you need it! 😄

File: Bookish/Controllers/MemberController.cs
Comment (emilypatsko) [inline Bookish/Controllers/MemberController.cs]: I would rename this method e.g. `AddMember`

File: Bookish/Controllers/MemberController.cs
Comment (emilypatsko) [inline Bookish/Controllers/MemberController.cs]: It doesn't make sense to me to return a view called ErrorMsg if the action is performed successfully...

File: Bookish/Views/Member/AddMember.cshtml
Comment (emilypatsko) [inline Bookish/Views/Member/AddMember.cshtml]: Shouldn't have empty paragraphs - you may want to read about using semantic HTML

File: Bookish/Views/Member/AddMember.cshtml
Comment (emilypatsko) [inline Bookish/Views/Member/AddMember.cshtml]: If this is a form to add a new member, it's submitting new data, and probably shouldn't be a GET method

File: Bookish/Views/Member/EditMember.cshtml
Comment (emilypatsko) [inline Bookish/Views/Member/EditMember.cshtml]: Similar here - forms are used for submitting data, which is typical of a POST request, not GET - GET requests don't submit any new data

File: Bookish/Views/Member/EditMember.cshtml
Comment (emilypatsko) [inline Bookish/Views/Member/EditMember.cshtml]: This doesn't need to be on a new line

File: Bookish/Views/Member/EditMember.cshtml
Comment (emilypatsko) [inline Bookish/Views/Member/EditMember.cshtml]: If this code isn't being used any more, delete it rather than commenting it out

File: Bookish/Views/Member/MemberCatalogue.cshtml
Comment (emilypatsko) [inline Bookish/Views/Member/MemberCatalogue.cshtml]: Inline style usage

File: Bookish/Views/Member/MemberCatalogue.cshtml
Comment (emilypatsko) [inline Bookish/Views/Member/MemberCatalogue.cshtml]: Rather than having a `List<Member>` as the model, I would create a new folder called `ViewModels`, and within it create a `MemberCatalogueViewModel`, which would contain all the information you'd want to show on this page. (e.g. it could include a `List<Member>` called `Members`)

File: Bookish/Views/Member/MemberQuery.cshtml
Comment (emilypatsko) [inline Bookish/Views/Member/MemberQuery.cshtml]: If this code isn't being used, I would delete it

File: Bookish/Views/Member/MemberQuery.cshtml
Comment (emilypatsko) [inline Bookish/Views/Member/MemberQuery.cshtml]: Inline style

File: Bookish/Views/Member/MemberQuery.cshtml
Comment (emilypatsko) [inline Bookish/Views/Member/MemberQuery.cshtml]: Empty paragraph

File: Bookish/Views/Member/MemberQueryByNameOrEmail.cshtml
Comment (emilypatsko) [inline Bookish/Views/Member/MemberQueryByNameOrEmail.cshtml]: Inline style

File: Bookish/Views/Member/MemberQueryByNameOrEmail.cshtml
Comment (emilypatsko) [inline Bookish/Views/Member/MemberQueryByNameOrEmail.cshtml]: Similar comment to earlier - create a ViewModel to hold the information you want to display on a particular view

File: Bookish/Views/Member/MemberSearch.cshtml
Comment (emilypatsko) [inline Bookish/Views/Member/MemberSearch.cshtml]: Doesn't need to be on new line

File: Bookish/Views/Member/MemberSearch.cshtml
Comment (emilypatsko) [inline Bookish/Views/Member/MemberSearch.cshtml]: Probably should be a POST method

**Review by emilypatsko — CHANGES_REQUESTED** [review]

File: Bookish/Controllers/MemberController.cs
Comment (Chestnut-lol) [reply Bookish/Controllers/MemberController.cs:171]: No... I tried using that but member.Books got lost somehow along the way, so we had to do it that way... (SingleOrDefault doesn't work either)

File: Bookish/Views/Member/AddMember.cshtml
Comment (Chestnut-lol) [reply Bookish/Views/Member/AddMember.cshtml]: Yeah ok I've changed them to line breaks.

File: Bookish/Controllers/MemberController.cs
Comment (Chestnut-lol) [reply Bookish/Controllers/MemberController.cs]: Yeah I'm hoping to change it after the merging because this model is called extensively across the whole project...

File: Bookish/Controllers/MemberController.cs
Comment (Chestnut-lol) [reply Bookish/Controllers/MemberController.cs]: Yeah sure will do.

File: Bookish/Controllers/MemberController.cs
Comment (Chestnut-lol) [reply Bookish/Controllers/MemberController.cs:171]: Nvm done now :-)

File: Bookish/Views/Member/AddMember.cshtml
Comment (Chestnut-lol) [reply Bookish/Views/Member/AddMember.cshtml]: Right, changed it to a POST method.

File: Bookish/Views/Member/EditMember.cshtml
Comment (Chestnut-lol) [reply Bookish/Views/Member/EditMember.cshtml]: Right - sure.

File: Bookish/Views/Member/EditMember.cshtml
Comment (Chestnut-lol) [reply Bookish/Views/Member/EditMember.cshtml]: ok

File: Bookish/Views/Member/EditMember.cshtml
Comment (Chestnut-lol) [reply Bookish/Views/Member/EditMember.cshtml]: ok

File: Bookish/Controllers/MemberController.cs
Comment (Chestnut-lol) [reply Bookish/Controllers/MemberController.cs:166]: no, I've deleted them.

File: Bookish/Controllers/MemberController.cs
Comment (Chestnut-lol) [reply Bookish/Controllers/MemberController.cs]: yeah sure. I've deleted them.

File: Bookish/Controllers/MemberController.cs
Comment (Chestnut-lol) [reply Bookish/Controllers/MemberController.cs]: I've changed it to another class MsgModel now, hopefully it will make more sense.

File: Bookish/Controllers/MemberController.cs
Comment (Chestnut-lol) [reply Bookish/Controllers/MemberController.cs]: ok, I have changed the code.

File: Bookish/Controllers/MemberController.cs
Comment (Chestnut-lol) [reply Bookish/Controllers/MemberController.cs]: ok, I have changed it.

File: Bookish/Views/Member/MemberCatalogue.cshtml
Comment (Chestnut-lol) [reply Bookish/Views/Member/MemberCatalogue.cshtml]: ok deleted.

File: Bookish/Views/Member/MemberCatalogue.cshtml
Comment (Chestnut-lol) [reply Bookish/Views/Member/MemberCatalogue.cshtml]: ok done.

File: Bookish/Views/Member/MemberQuery.cshtml
Comment (Chestnut-lol) [reply Bookish/Views/Member/MemberQuery.cshtml]: ok.

File: Bookish/Views/Member/MemberQuery.cshtml
Comment (Chestnut-lol) [reply Bookish/Views/Member/MemberQuery.cshtml]: ok, deleted.

File: Bookish/Views/Member/MemberQueryByNameOrEmail.cshtml
Comment (Chestnut-lol) [reply Bookish/Views/Member/MemberQueryByNameOrEmail.cshtml]: ok. deleted.

File: Bookish/Views/Member/MemberQuery.cshtml
Comment (Chestnut-lol) [reply Bookish/Views/Member/MemberQuery.cshtml]: ok deleted.

File: Bookish/Views/Member/MemberQueryByNameOrEmail.cshtml
Comment (Chestnut-lol) [reply Bookish/Views/Member/MemberQueryByNameOrEmail.cshtml]: ok done.

File: Bookish/Views/Member/MemberSearch.cshtml
Comment (Chestnut-lol) [reply Bookish/Views/Member/MemberSearch.cshtml]: ok done.

File: Bookish/Views/Member/MemberSearch.cshtml
Comment (Chestnut-lol) [reply Bookish/Views/Member/MemberSearch.cshtml]: yes, changed.

File: Bookish/Controllers/MemberController.cs
Comment (Chestnut-lol) [reply Bookish/Controllers/MemberController.cs:74]: I'll rename it after merging with Blanca? Just because this method might have been used by her too.

**Pull request merged**
